x/datadeal/types/keys.go

```diff
@@ -67,8 +67,12 @@ func GetDataVerificationQueueKeyByTimeKey(endTime time.Time) []byte {
 	return append(DataVerificationQueueKey, sdk.FormatTimeBytes(endTime)...)
 }
 
-func SplitDataVerificationQueueKey(key []byte) (uint64, string) {
-	return sdk.BigEndianToUint64(key[1+lenTime+1 : 1+lenTime+1+8]), string(key[1+lenTime+1+8+1:])
+func SplitDataVerificationQueueKey(key []byte) (*time.Time, uint64, string, error) {
```

Comment:
I think we can share this function for verification and delivery.

Reply:
Fixed in [fe82c2f](https://github.com/medibloc/panacea-core/pull/496/commits/fe82c2f0adcfb869824d35615ed986ce3d46e973). I changed the name to `SplitDataQueueKey`, but I'm not sure the name is proper.